crates/nargo/src/cli/prove_cmd.rs

```diff
@@ -119,3 +84,53 @@ fn process_abi_with_input(
     }
     Ok(solved_witness)
 }
+
+pub fn prove_hlp(
```

Comment:
Lets rename prove_hlp to `prove_with_path`

- Nit: can you rename prg_dir to program_dir?

Reply:
Also same suggestion about using AsRef<Path> , PathBuf ideally should be used when we are going to modify the path again